### DISCUSSION TITLE
Store attribute is `undefined` after using `$store.foo.some_method` without parentheses

### DISCUSSION THREAD

**Discussion by peterhijma** [alpinejs/alpine#1593] [1. Help]
I've got some store defined, called `foo` with an attribute called `bar` and an **async** method called `bazMethod`.

I want to use the event inside `bazMethod`, so I don't use parentheses.

```
<button @click="$store.foo.bazMethod">
```

I can `console.log` the event inside `bazMethod`, so that worked. But... attribute `bar` is now `undefined` ! :(

Any idea?

**Comment by ryangjchandler** [comment]
@peterhijma I think the issue here is that Alpine is binding the `this` context to the component's data, instead of the store's data. I'd recommend using `$store.foo.bazMethod()` for now.

If you need the event, you can pass it through as: `@click="$store.foo.bazMethod($event)"`.

**Reply by peterhijma** [reply]
Yeah, I'm using that now indeed! Just wondered if I did something wrong or that this might be a potential bug :)

Thanks for the quick reply.
